### ISSUE TITLE
UWSim with ROS Indigo - in-water camera view is very 'noisey'

### ISSUE THREAD

**Issue by ghost** [uji-ros-pkg/underwater_simulation#42]
Hi,

I have built UWSim and ROS Indigo from source. However when I run UWSim without using --disableShader the camera view looks terrible under the water. Outside the water it looks fine.

With --disableShader the surface of the water looks solid and the water appears invisible.

I wonder if anyone else has had any similar issues.

**Comment by marioprats** [comment]
Same here, not sure what the problem is...

**Comment by ghost** [comment]
Here is a screenshot of what I can see:

http://i61.tinypic.com/qrhu09.png

**Comment by perezsolerj** [comment]
Hi everyone,

This seems to be a driver-related issue. I guess new default linux driver supports some kind of shaders, but osgOcean godrays effects are causing trouble. So now instead of causing a segmentation fault it runs with this bug. I would recommend to update your drivers and see if it works (propietary drivers).

I made some tests in order to solve this. Running UWSim with --disableShaders solves the issue, after experimenting a little bit I realized that disabling shaders and godRays effects it's enough (Lines 116 and 137 on SceneBuilder).

So i run both configurations with --vv option and the only difference between them (after solving some conversion errors on default_shader) is:

glValidateProgram FAILED "ocean_surface" id=18 contextID=0
infolog:
Texture unit 0 is accessed both as sampler2D and samplerCube

I guess if we solve this, we will solve the noisey behaviour with the default linux driver.
